core/src/test/java/com/gentics/mesh/core/field/s3binary/S3BinaryFieldEndpointTest.java

```diff
@@ -0,0 +1,269 @@
+package com.gentics.mesh.core.field.s3binary;
+
+import com.gentics.mesh.FieldUtil;
+import com.gentics.mesh.core.db.Tx;
+import com.gentics.mesh.core.field.AbstractFieldEndpointTest;
+import com.gentics.mesh.core.rest.node.NodeCreateRequest;
+import com.gentics.mesh.core.rest.node.NodeResponse;
+import com.gentics.mesh.core.rest.node.field.s3binary.S3RestResponse;
+import com.gentics.mesh.core.rest.schema.S3BinaryFieldSchema;
+import com.gentics.mesh.core.rest.schema.SchemaVersionModel;
+import com.gentics.mesh.core.rest.schema.impl.S3BinaryFieldSchemaImpl;
+import com.gentics.mesh.parameter.impl.ImageManipulationParametersImpl;
+import com.gentics.mesh.rest.client.MeshBinaryResponse;
+import com.gentics.mesh.test.context.MeshTestSetting;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static com.gentics.mesh.test.ClientHelper.call;
+import static com.gentics.mesh.test.TestDataProvider.PROJECT_NAME;
+import static com.gentics.mesh.test.context.AWSTestMode.MINIO;
+import static org.junit.Assert.*;
+
+@MeshTestSetting(awsContainer = MINIO, startServer = true)
+public class S3BinaryFieldEndpointTest extends AbstractFieldEndpointTest {
+
+    private static final String FIELD_NAME = "s3";
+    private static final String uploadBody = "{\n" +
```

Comment:
This approach is not applicable to the MeshRestClient - there should be no externally constructed JSON body, as it is error-prone. Every parameter should come either as a separate Java method parameter or a logically merged parameter class. See [the binary client](https://github.com/gentics/mesh/blob/feature/Mesh-S3-Support/rest-client/src/main/java/com/gentics/mesh/rest/client/method/NodeBinaryFieldClientMethods.java#L37) for an example.